src/main/java/org/apache/commons/crypto/cipher/OpensslCipher.java

```diff
@@ -80,19 +83,25 @@ public Properties getProperties() {
    * Initializes the cipher with mode, key and iv.
    * @param mode {@link #ENCRYPT_MODE} or {@link #DECRYPT_MODE}
    * @param key crypto key for the cipher
-   * @param iv Initialization vector for the cipher
+   * @param params the algorithm parameters
    * @throws IOException if cipher initialize fails
    */
   @Override
-  public void init(int mode, byte[] key, byte[] iv) {
+  public void init(int mode, Key key, AlgorithmParameterSpec params) {
     Utils.checkNotNull(key);
-    Utils.checkNotNull(iv);
+    Utils.checkNotNull(params);
 
     int cipherMode = Openssl.DECRYPT_MODE;
     if (mode == ENCRYPT_MODE)
       cipherMode = Openssl.ENCRYPT_MODE;
-
-    cipher.init(cipherMode, key, iv);
+    byte[] ivBytes;
+    if (params instanceof IvParameterSpec) {
+      ivBytes = ((IvParameterSpec) params).getIV();
+    } else {
+      //other AlgorithmParameterSpec such as GCMParameterSpec is not supported now.
```

Comment:
@jerrychenhf, agree with you. It would be better to throw an java exception here.  fixed.